autoblocks/_impl/testing/evaluators/battle.py

```diff
@@ -23,37 +23,130 @@ class BattleResponse:
+                    You must provide one answer based on your subjective view and provide a reason for your answer.
+
+                    Always output in the following JSON format:
+
+                    {
+                      "reason": "This is the reason.",
+                      "result": "0" | "1" | "2"
+                    }"""
+                ),
+            ),
+            dict(
+                role="user",
+                content=dedent(
+                    f"""[Criteria]
+                    {criteria}
+
+                    [Baseline]
+                    {baseline}
+
+                    [Challenger]
+                    {challenger}"""
+                ),
+            ),
+        ],
+    )
+
+    raw_content = response.choices[0].message.content
+    if not raw_content:
+        raise ValueError("No content was returned from OpenAI for the Battle evaluator")
+
+    parsed_content = json.loads(raw_content.strip())
+    battle_response = BattleResponse(
+        result=parsed_content["result"],
+        reason=parsed_content["reason"],
+    )
+    score: float = 0
+    if battle_response.result == "2":
+        score = 1
+    elif battle_response.result == "0":
+        score = 0.5
+
+    return Evaluation(
+        score=score,
+        threshold=threshold,
+        metadata={
+            "reason": battle_response.reason,
+            "baseline": baseline,
+            "challenger": challenger,
+            "criteria": criteria,
+        },
+    )
+
+
+class ManualBattle(BaseTestEvaluator, abc.ABC):
     """
     The Battle evaluator compares two responses based on a given criteria.
-    If the challenger wins, the challenger becomes the new baseline automatically.
-    You can override this behavior by passing in a baseline_mapper and handling the baseline yourself.
     """
 
-    id = "battle"
     threshold = Threshold(gte=0.5)  # consider a tie as passing
 
-    def __init__(
-        self,
-        # Criteria to be used in the battle
-        criteria: str,
-        # Map your output to a string for comparison
-        output_mapper: Callable[[OutputType], str],
-        # Optional baseline_mapper allows you to use a baseline from your test case
-        # Instead of having Autoblocks automatically track it from the output
-        baseline_mapper: Optional[Callable[[TestCaseType], str]] = None,
-    ):
-        super().__init__()
-        self.output_mapper = output_mapper
-        self.baseline_mapper = baseline_mapper
-        self.criteria = criteria
+    @property
+    @abc.abstractmethod
+    def criteria(self) -> str:
+        pass
 
-    async def get_baseline(self, test_id: str, test_case: TestCaseType) -> Optional[str]:
-        # if the user passed in a baseline_mapper, we use that
-        if self.baseline_mapper:
-            return self.baseline_mapper(test_case)
+    @abc.abstractmethod
+    def baseline_mapper(self, test_case: Any) -> str:
```

Comment:
you can keep the generics so that this can be typed `test_case: TestCaseType`

we only had to drop it for `evaluate_test_case` b/c we needed to support kwargs for hook results